code/datums/components/equipment_fault.dm

```diff
@@ -454,3 +454,75 @@ TYPEINFO(/datum/component/equipment_fault)
 			flame_turf.hotspot_expose(T0C + 50, 50)
 			playsound(flame_turf, 'sound/effects/gust.ogg', 50, FALSE)
 			O.visible_message(SPAN_NOTICE("An ember flies out of [O]."))
+
+///leaks chemicals to nearby tiles
+/datum/component/equipment_fault/leaky
+	///base/current probability to spawn fluids on this process tick
+	var/current_prob = 5
+	///increase in probability per process tick
+	var/static/prob_raise = 5
+	///list of reagents IDs to leak
+	var/list/reagent_list = list("carbon", "copper", "iron", "nickel", "oil")
+	var/static/list/sounds = list(
+		'sound/machines/vending_dispense_small.ogg',
+		'sound/machines/decompress.ogg',
+		'sound/effects/splort.ogg',
+		'sound/effects/zzzt.ogg',
+	)
+
+/datum/component/equipment_fault/leaky/ef_process(obj/machinery/M, mult)
+	if(probmult(current_prob))
+		src.ef_perform_fault(M)
+		src.current_prob = initial(src.current_prob)
+	else
+		src.current_prob += src.prob_raise
+
+/datum/component/equipment_fault/leaky/ef_perform_fault(obj/O)
+	if(..())
+		var/target_dir = pick(alldirs)
+		var/turf/simulated/floor/T = get_step(O.loc, target_dir)
+		if(!istype(T))
+			T = O.loc
+		else
+			var/obj/effects/spray/spray = new(T)
+			SPAWN(1 SECOND) qdel(spray)
+			spray.set_dir(target_dir)
+		playsound(O, pick(sounds), 50, 2)
+		O.visible_message(SPAN_NOTICE("Some of the contents of [O] leaks onto the floor."))
+
+		var/datum/reagents/temp_fluid_reagents = new /datum/reagents(5)
+		temp_fluid_reagents.add_reagent(pick(src.reagent_list), 5)
+		T.fluid_react(temp_fluid_reagents, temp_fluid_reagents.total_volume)
+
+/datum/component/equipment_fault/leaky/chemical
+	reagent_list = list(
```

Comment:
Why not just pass in this list from the parent? Otherwise it will break on varedited chem dispensers or if any of these lists change in the future.

Reply:
added the list of reagents as an initialization argument to the component and removed the subclasses. the parent reagent dispenser now adds the component with the current machine's dispensable reagents.